### PR TITLE
rgw/notifications: dont send notifications if op has failed

### PR DESCRIPTION
**Pull request by yuvalif**
Fixes: https://tracker.ceph.com/issues/63855

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
teuthology passing: https://pulpito.ceph.com/yuvalif-2023-12-20_09:23:38-rgw:notifications-wip-yuval-63855-distro-default-smithi/

**Comment by yuvalif** [comment]
jenkins test make check

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -7266,7 +7277,9 @@ void RGWDeleteMultiObj::handle_individual_object(const rgw_obj_key& o, optional_
   if (op_ret == -ENOENT) {
     op_ret = 0;
   }
-
+  if (op_ret < 0) {
+    return;
+  }
```
Comment (cbodley) [inline src/rgw/rgw_op.cc:7282]: `RGWDeleteMultiObj::handle_individual_object()` needs to communicate this error to `send_partial_response()`

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
teuthology: https://pulpito.ceph.com/yuvalif-2024-01-09_15:23:42-rgw-wip-yuval-63855-distro-default-smithi/
* multisite is mostly failing: http://qa-proxy.ceph.com/teuthology/yuvalif-2024-01-09_15:23:42-rgw-wip-yuval-63855-distro-default-smithi/7509564/teuthology.log
* 2 failures with: 
```
ceph --cluster ceph tell 10.c deep_scrub
no valid command found
```
which seems unrelated

**Comment by yuvalif** [comment]
> teuthology: https://pulpito.ceph.com/yuvalif-2024-01-09_15:23:42-rgw-wip-yuval-63855-distro-default-smithi/
> 
> * multisite is mostly failing: http://qa-proxy.ceph.com/teuthology/yuvalif-2024-01-09_15:23:42-rgw-wip-yuval-63855-distro-default-smithi/7509564/teuthology.log
> * 2 failures with:
> 
> ```
> ceph --cluster ceph tell 10.c deep_scrub
> no valid command found
> ```
> 
> which seems unrelated

failures are same as baseline:
https://pulpito.ceph.com/yuriw-2024-01-12_15:23:49-rgw-main-distro-default-smithi/

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
> ceph --cluster ceph tell 10.c deep_scrub

tracked in https://tracker.ceph.com/issues/63967

**Pull request merged**